src/StreamFactoryInterface.php

```diff
@@ -9,15 +9,11 @@
     /**
      * Create a new stream from a resource or a string.
      *
-     * If the argument is a resource, it MUST be readable and SHOULD be seekable. It MAY be writable.
-     * If the argument is a string, a temporary resource will be created that is writable and seekable.
-     * 
-     * If the arugment is a string it will be interpided as the content of the stream. File names or 
-     * file paths are not supported. 
+     * String content will be used as the body of the stream. Files paths are not supported.
```

Comment:
I agree with your points and I think you are **technically correct**. But I think you are wrong when considering current implementations and Im not sure multiple stream interfaces would be easy to use for "most users". 

I've made my arguments before in #4 and #14 so I do not repeat them. 

Is there a middle ground here? Would it make sense to modify @mindplay-dk's suggestion by *adding* an interface?

```php
FileStreamFactory::createFileStream(string $path, string $mode = "r");
ResourceStreamFactory::createResourceStream(resource $handle);
TempStreamFactory::createTempStream(string $content);
MemoryStreamFactory::createMemoryStream(string $content);

// This is our current interface. The implementation will decide what type of stream this is.
CommonStreamFactory::createStream(string $content);
```